### PR TITLE
Fixes #21418: Campaign scheduler seems to not look for existing campaign at boot

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/21418

### CODE REVIEW

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/campaigns/DataTypes.scala [hunk 1]
```diff
@@ -53,15 +53,30 @@ trait Campaign {
...
+    id : CampaignId
   , name : String
   , description : String
   , status : CampaignStatus
   , schedule : CampaignSchedule
   , duration: Duration
 )
 
-case class CampaignId (value : String, rev: Revision = GitVersion.DEFAULT_REV)
+case class CampaignId (value : String, rev: Revision = GitVersion.DEFAULT_REV) {
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/campaigns/DataTypes.scala:64]: I'm not sure we need a revision for campaign id, but if we need one for futur-proveness, it needs to be serialized with revision

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

**Comment by VinceMacBuche** [comment]
https://github.com/fanf/rudder/pull/14

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

**Review by VinceMacBuche — APPROVED** [review]

**Comment by VinceMacBuche** [comment]
OK, squash merging this PR

**Comment by VinceMacBuche** [comment]
PR rebased

**Comment by VinceMacBuche** [comment]
OK, merging this PR

**Pull request merged**